src/samples.jl

```diff
@@ -0,0 +1,179 @@
+
+module MAGMA
+
+export magmaInit, magmaFinalize, gesvd!
+
+# low-level wrappers of the MAGMA library
+include("common.jl")
+
+
+# magma_init
+function magmaInit()
+	ccall((:magma_init, libmagma),Cint,())
+end
+
+# magma_finalize
+function magmaFinalize()
+	ccall((:magma_finalize, libmagma),Cint,())
+end
+
+# magma_cgesvd wrapper
+function magmaCgesvd(jobu, jobvt, m, n, A, lda, s, U,
+	ldu, VT, ldvt, work, lwork, rwork, info)
+    ccall((:magma_cgesvd, libmagma),
+               Cint,
+               (Cint, Cint, Cint, Cint, PtrOrCuPtr{ComplexF32},
+			    Cint, PtrOrCuPtr{Float32}, PtrOrCuPtr{ComplexF32},
+				Cint, PtrOrCuPtr{ComplexF32}, Cint,
+				PtrOrCuPtr{ComplexF32}, Cint, PtrOrCuPtr{Float32},
+				PtrOrCuPtr{Cint}),
+                jobu, jobvt, m, n, A, lda, s, U, ldu, VT, ldvt,
+			    work, lwork, rwork, info)
+end
+
+# magma_dgesvd wrapper
+function magmaDgesvd(jobu, jobvt, m, n, A, lda, s, U,
+	 ldu, VT, ldvt, work, lwork, info)
+    ccall((:magma_dgesvd, libmagma),
+               Cint,
+               (Cint, Cint, Cint, Cint, PtrOrCuPtr{Float64},
+			    Cint, PtrOrCuPtr{Float64}, PtrOrCuPtr{Float64},
+				Cint, PtrOrCuPtr{Float64}, Cint,
+				PtrOrCuPtr{Float64}, Cint, PtrOrCuPtr{Cint}),
+                jobu, jobvt, m, n, A, lda, s, U, ldu, VT, ldvt,
+			    work, lwork, info)
+end
+
+# magma_sgesvd wrapper
+function magmaSgesvd(jobu, jobvt, m, n, A, lda, s, U,
+	 ldu, VT, ldvt, work, lwork, info)
+    ccall((:magma_sgesvd, libmagma),
+               Cint,
+               (Cint, Cint, Cint, Cint, PtrOrCuPtr{Float32},
+			    Cint, PtrOrCuPtr{Float32}, PtrOrCuPtr{Float32},
+				Cint, PtrOrCuPtr{Float32}, Cint, PtrOrCuPtr{Float32},
+				Cint, Ptr{Cint}),
+                jobu, jobvt, m, n, A, lda, s, U, ldu, VT, ldvt,
+				work, lwork, info)
+end
+
+# magma_cgesvd wrapper
+function magmaZgesvd(jobu, jobvt, m, n, A, lda, s, U,
+	 ldu, VT, ldvt, work, lwork, rwork, info)
+    ccall((:magma_zgesvd, libmagma),
+               Cint,
+               (Cint, Cint, Cint, Cint, PtrOrCuPtr{ComplexF64},
+			    Cint, PtrOrCuPtr{Float64}, PtrOrCuPtr{ComplexF64},
+				Cint, PtrOrCuPtr{ComplexF64}, Cint, PtrOrCuPtr{ComplexF64},
+				Cint, PtrOrCuPtr{Float64}, PtrOrCuPtr{Cint}),
+                jobu, jobvt, m, n, A, lda, s, U, ldu, VT, ldvt,
+			    work, lwork, rwork, info)
+end
+
+
+# wrappers of the low-level MAGMA functions
+for (function_name, element_type, singular_value_type) in
+	((:magmaCgesvd,:ComplexF32,:Float32),
```

Comment:
we only have a svd wrapper here, the whole point of `eval` is to generate code with generic writing, you should generate the wrapper for different functions

```
for (flag, ElType, SVTyle) in [(:C, :ComplexF32, :Float32), (:Z, :ComplexF64, :Float64)]
        magma_function = Symbol(:magma, flag, :gesvd)
        function $magma_function(blabla)
              # declaration
        end
end
```

Reply:
This can be more generic, if you only iterate the precision type

```julia
for (flag, PType) in [(:C, Float32), blbla]
     ElType = Complex{PType}
      #....
end
```
```